src/utils.js

```diff
@@ -224,6 +224,17 @@ function isDOMElement(element) {
   }
 }
 
+/**
+ * React Native sets `navigator.product` to a constant `ReactNative`.
+ * We can use this to determine if the target is React Native or not.
+ * See https://github.com/facebook/react-native/issues/10881 for more info.
+ * See https://github.com/facebook/react-native/blob/70c359000a2df091c3939f4c19db6024af992d43/Libraries/Core/InitializeCore.js#L194-L195 for more info.
+ * @return {Boolean} whether or not the platform is React native
+ */
+function isReactNative() {
+  return navigator != null && navigator.product === 'ReactNative'
```

Comment:
Can this be `preval`ed?

Reply:
I think the answer is no, since this needs to exist at runtime. Maybe there's a better alternative to use here that could help us cut down on some function calls.